source/lmp/fix_dplr.cpp

```diff
@@ -157,6 +192,48 @@ void FixDPLR::init() {
   // 	 << vv[ii][2] << " "
   // 	 << endl;
   // }
+  // check variables
+
+  if (xstr) {
+    xvar = input->variable->find(xstr);
+    if (xvar < 0)
+      error->all(FLERR, "Variable {} for x-field in fix {} does not exist",
+                 xstr, style);
+    if (input->variable->equalstyle(xvar))
+      xstyle = EQUAL;
+    else
+      error->all(FLERR, "Variable {} for x-field in fix {} is invalid style",
+                 xstr, style);
+  }
+
+  if (ystr) {
+    yvar = input->variable->find(ystr);
+    if (yvar < 0)
+      error->all(FLERR, "Variable {} for y-field in fix {} does not exist",
+                 ystr, style);
+    if (input->variable->equalstyle(yvar))
+      ystyle = EQUAL;
+    else
+      error->all(FLERR, "Variable {} for y-field in fix {} is invalid style",
+                 ystr, style);
+  }
+
+  if (zstr) {
+    zvar = input->variable->find(zstr);
+    if (zvar < 0)
+      error->all(FLERR, "Variable {} for z-field in fix {} does not exist",
+                 zstr, style);
+    if (input->variable->equalstyle(zvar))
+      zstyle = EQUAL;
+    else
+      error->all(FLERR, "Variable {} for z-field in fix {} is invalid style",
+                 zstr, style);
+  }
```

Comment:
Is it possible to write a loop? Here x, y, z are exactly the same.

Reply:
It is possible, but I do not recommend doing that. This part is almost the same as that in LAMMPS's fix_efield.cpp. Keeping them the same makes it easier to maintain.

Reply:
I see.